### DISCUSSION TITLE
Quick Start Guide?

### DISCUSSION THREAD

**Discussion by pm3003** [dionysius/vaultwarden-deb#9] [Q&A] [answered]
So I installedthe packages (vaultwarden and vaultwarden-web-vault)... and I have no idea what to do next.
The vaultwarden service file gave me the adress of the default config file (vim /etc/default/vaultwarden) and I was able to launch it successfully. I can see it listening on port 8000. 

However, it won't take connections from my home network, and I hve no idea how to start the web interface.
I've noticed a bunch of html and JS files in /usr/share/vaultwarden-web-vault, but I have no idea what I'm supposed to do with them. Should I write a service that npm runs in the folder, or install a web server and copy the files to www-data?

How do I setup a user to be able to connect?

Any help would be appreciated.

**Comment by dionysius** [comment, marked as answer]
You are right, a little quick start instruction would be best placed in a man page, which does not exist yet. Their [wiki for configuration options](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview#configuration-options) is also a good link to point users to (you can ignore all the env variable loading stuff, it is already handled for you).

I currently assume people to find this config file - as I try to comply as strictly as possible with the [debian policy manual](https://www.debian.org/doc/debian-policy/index.html) which standardise locations of files - and would be able to start configuring the service - and IMO the upstream config file is very detailed.

[I already "complained"](https://github.com/dani-garcia/vaultwarden/issues/3077) about this issue. You might have missed the comment `Uncomment any of the following lines to change the defaults` at the beginning of the file. It listens by default on `127.0.0.1` while it does not show that in the file. In practice I agree on it being localhost, as you **really want** to configure the service before making it to listen to all interfaces (`0.0.0.0`) or a public interface (specific address). It's usually the admins discretion on which address a server listens and maintainer defaults are a bit cautious.

So no, you don't need to do any npm stuff, everyting is ready out-of-the-box. The section you want to change is:
```
## Rocket specific settings
## See https://rocket.rs/v0.4/guide/configuration/ for more details.
# ROCKET_ADDRESS=0.0.0.0
# ROCKET_PORT=80  # Defaults to 80 in the Docker images, or 8000 otherwise.
# ROCKET_WORKERS=10
# ROCKET_TLS={certs="/path/to/certs.pem",key="/path/to/key.pem"}
```

And restart the service with `systemctl restart vaultwarden`. Once you've changed the `ROCKET_ADDRESS` and maybe `ROCKET_PORT` you should be able to access it over your local network. But also have a look at the other sections, e.g. who is able to create accounts etc.

Additionally: It's in your discretion if you want to put a [reverse-proxy](https://en.wikipedia.org/wiki/Reverse_proxy) in front of it for e.g. [TLS termination](https://en.wikipedia.org/wiki/TLS_termination_proxy) instead of doing it directly with vaultwardens rocket options, because in such case `127.0.0.1` would be the correct listener address if it is on the same host.

**Reply by pm3003** [reply]
Thank you very much, I agree with the location of the /etc/default config file (it's been a while since I saw one there) and the fact that it starts while listening to localhost only (in principle).
I'm not familiar with Rocket so I first assume at first it was some specific accessory service.
